tide/master/PixelStreamWindowManager.cpp

```diff
@@ -191,24 +191,41 @@ void PixelStreamWindowManager::registerEventReceiver(
     success->set_value(false);
 }
 
-void PixelStreamWindowManager::updateStreamDimensions(deflect::FramePtr frame)
+void PixelStreamWindowManager::_updateWindowSize(ContentWindowPtr window,
+                                                 const QSize size)
 {
-    const auto size = frame->computeDimensions();
-
-    auto window = getWindow(frame->uri);
-    if (!window)
-        return;
+    bool emitOpen = false;
 
     // External streamers might not have reported an initial size yet
     if (window->getContent()->getDimensions().isEmpty())
     {
+        emitOpen = true;
+
         const auto target = ContentWindowController::Coordinates::STANDARD;
         ContentWindowController controller{*window, _displayGroup, target};
         controller.resize(size, CENTER);
     }
+
+    if (size == window->getContent()->getDimensions())
+        return;
+
     window->getContent()->setDimensions(size);
     if (window->isFocused())
         DisplayGroupController{_displayGroup}.updateFocusedWindowsCoordinates();
+
+    if (emitOpen)
+        emit externalStreamOpening(window->getContent()->getURI());
+}
+
+void PixelStreamWindowManager::updateStreamDimensions(deflect::FramePtr frame)
+{
+    const auto size = frame->computeDimensions();
+
+    auto window = getWindow(frame->uri);
+    if (!window)
+        return;
+
+    _updateWindowSize(window, size);
```

Comment:
Not critical but frame->computeDimensions() should normally go after the early-out for the window not found.
Or could even be a two-liner:
```
if (auto window = getWindow(frame->uri))
    _updateWindowSize(window, frame->computeDimensions());
```